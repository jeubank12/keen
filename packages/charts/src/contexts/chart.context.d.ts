import React from 'react';
import { Dimension, Margins, Theme, ScaleSettings } from '../types';
export declare type ChartContextType = {
  theme: Theme;
  svgDimensions: Dimension;
  margins: Margins;
  xScaleSettings: ScaleSettings;
  yScaleSettings: ScaleSettings;
  data: object[];
};
declare const ChartContext: React.Context<{}>;
export default ChartContext;
