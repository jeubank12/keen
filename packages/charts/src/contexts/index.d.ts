import ChartContext, { ChartContextType } from './chart.context';
import ScaleContext, { ScaleContextType } from './scale.context';
export { ChartContext, ScaleContext, ScaleContextType, ChartContextType };
