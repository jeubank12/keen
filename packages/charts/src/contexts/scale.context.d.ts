import React from 'react';
import { ScaleBand, ScaleLinear } from 'd3-scale';
export declare type ScaleContextType = {
  xScale: ScaleBand<string> | ScaleLinear<number, number>;
  yScale: ScaleLinear<number, number> | ScaleBand<string>;
};
declare const ScaleContext: React.Context<{}>;
export default ScaleContext;
