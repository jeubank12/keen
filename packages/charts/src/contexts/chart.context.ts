import React from 'react';
import { ScaleSettings } from '@keen.io/charts-utils';

import { Dimension, Margins, Theme } from '../types';

export type ChartContextType = {
  theme: Theme;
  svgDimensions: Dimension;
  margins: Margins;
  xScaleSettings: ScaleSettings;
  yScaleSettings: ScaleSettings;
  data: Record<string, any>[];
};

const ChartContext = React.createContext({});

export default ChartContext;
