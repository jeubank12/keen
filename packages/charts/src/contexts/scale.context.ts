import React from 'react';
import { ScaleBand, ScaleLinear } from 'd3-scale';

export type ScaleContextType = {
  xScale: ScaleBand<string> | ScaleLinear<number, number>;
  yScale: ScaleLinear<number, number> | ScaleBand<string>;
};

const ScaleContext = React.createContext({});

export default ScaleContext;
