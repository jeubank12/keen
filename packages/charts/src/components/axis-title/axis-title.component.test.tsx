import React from 'react';
import { mount } from 'enzyme';

import AxisTitle from './axis-title.component';

import { ChartContext } from '../../contexts';

const x = { value: 'xTitle' };
const y = { value: 'yTitle' };

const setup = (overProps: any = {}) => {
  const props = {
    x,
    y,
    ...overProps,
  };

  const context = {
    margins: { top: 10, bottom: 10, left: 10, right: 10 },
    svgDimensions: { width: 100, height: 100 },
    theme: {
      axisX: {
        tickSize: 10,
        tickPadding: 10,
        labels: {
          typography: {
            fontSize: 10,
          },
        },
      },
      axisY: {
        tickSize: 10,
        tickPadding: 10,
        labels: {
          typography: {
            fontSize: 10,
          },
        },
      },
    },
  };

  const wrapper = mount(
    <ChartContext.Provider value={context}>
      <svg>
        <AxisTitle {...props} />
      </svg>
    </ChartContext.Provider>
  );

  return { wrapper, props };
};

describe('<AxisTitle />', () => {
  it('should render AxisTitle', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render AxisTitle values', () => {
    const { wrapper } = setup();
    expect(
      wrapper
        .find('text')
        .first()
        .text()
    ).toEqual(x.value);
    expect(
      wrapper
        .find('text')
        .last()
        .text()
    ).toEqual(y.value);
  });
});
