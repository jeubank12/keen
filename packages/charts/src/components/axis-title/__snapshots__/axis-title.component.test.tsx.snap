// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AxisTitle /> should render AxisTitle 1`] = `
<svg>
  <AxisTitle
    x={
      Object {
        "value": "xTitle",
      }
    }
    y={
      Object {
        "value": "yTitle",
      }
    }
  >
    <Title
      axisTitle={
        Object {
          "value": "xTitle",
        }
      }
      orientation="horizontal"
    >
      <text
        fill="currentColor"
        style={Object {}}
        textAnchor="middle"
        x={50}
        y={140}
      >
        xTitle
      </text>
    </Title>
    <Title
      axisTitle={
        Object {
          "value": "yTitle",
        }
      }
      orientation="vertical"
    >
      <text
        fill="currentColor"
        style={
          Object {
            "transform": "rotate(-90deg)",
          }
        }
        textAnchor="middle"
        x={-50}
        y={-50}
      >
        yTitle
      </text>
    </Title>
  </AxisTitle>
</svg>
`;
